mlflow/models/model_config.py

```diff
@@ -11,8 +11,56 @@
 
 class ModelConfig:
     """
-    ModelConfig used in code to read a YAML configuration file, and this configuration file can be
-    overridden when logging a model.
+    ModelConfig used in code to read a YAML configuration file or a dictionary.
+    When invoking the ModelConfig locally in a model file, development_config can be passed in
+    which would be used as configuration for the model. But this development_config configuration
+    file will be overridden when logging a model.
+
+    Args:
+        development_config: Path to the YAML configuration file or a dictionary containing the
+                            configuration. If the configuration is not provided, an error is raised.
+
+    .. code-block:: python
+        :caption: Example usage in model code
+
+        from mlflow.models import ModelConfig
+
+        # Load the configuration from a file
+        config = ModelConfig(development_config="config.yaml")
+        print(config.get("key1"))
+
+        # Load the configuration from a dictionary
+        config = ModelConfig(development_config={"key1": "value1"})
+        print(config.get("key1"))
+
+    .. code-block:: python
+        :caption: Example to pass in config file when logging the model
+
+        # agent.py
+        import mlflow
+        from mlflow.models import ModelConfig
+
+        config = ModelConfig(development_config={"key1": "value1"})
+
+
+        class TestModel(mlflow.pyfunc.PythonModel):
+            def predict(self, context, model_input, params=None):
+                return config.get("key1")
+
+
+        mlflow.models.set_model(TestModel())
+
+
+        # deploy.py
+        model_config = {"key1": "value2"}
+        mlflow.pyfunc.save_model(
```

Comment:
can we use `log_model` instead? `save_model` only logs the model locally and doesn't update it as an artifact.

Reply:
```suggestion
        with mlflow.start_run():
            mlflow.pyfunc.log_model(
```